src/core/namespace.c

```diff
@@ -519,6 +539,35 @@ static int append_extensions(
                                         "No matching entry in .v/ directory %s found.",
                                         m->source);
 
+                /* If this is a dm-verity image, check for duplicates, as Overlayfs rejects them as
+                 * explained above. */
+                r = verity_settings_load(&verity, result.path, /* root_hash_path= */ NULL, /* root_hash_sig_path= */ NULL);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to check verity root hash of %s: %m", result.path);
+
+                if (verity.root_hash_size > 0) {
+                        bool dup = false;
+
+                        FOREACH_ARRAY(v, verities, n_verities)
+                                if (memcmp_nn(v->root_hash,
+                                              v->root_hash_size,
+                                              verity.root_hash,
+                                              verity.root_hash_size) == 0) {
+                                        dup = true;
+                                        break;
+                                }
```

Comment:
this function is *waaaaaaaay* too large already. please move this loop out, i.e. call this verities_find() or so.